### PR TITLE
Make changes for indv roles menu (menu still in progress)

### PR DESCRIPTION
**Pull request by RXWatcher1**
1) removed static ips
2)  moved directory creation to each role
3) added ansible-toolbox to common role

### CODE REVIEW

File: roles/nginx-proxy/tasks/main.yml [hunk 1]
```diff
@@ -7,6 +7,11 @@
   shell: "id -g {{user}}"
   register: gid
 
+- name: Ensure rclone folders exists
```
Comment (l3uddz) [inline roles/nginx-proxy/tasks/main.yml]: this should say nginx-proxy

File: roles/nzbhydra/tasks/main.yml [hunk 2]
```diff
@@ -12,6 +12,10 @@
     name: nzbhydra
     state: absent
 
+- name: Create directories
+  file: "path={{item}} state=directory mode=0775 owner={{user}} group={{user}}"
+  with_items:
+    - /opt/nzbhydra
```
Comment (l3uddz) [inline roles/nzbhydra/tasks/main.yml]: missing an empty line below here

File: roles/organizr/tasks/main.yml [hunk 3]
```diff
@@ -12,6 +12,10 @@
     name: organizr
     state: absent
 
+- name: Create directories
+  file: "path={{item}} state=directory mode=0775 owner={{user}} group={{user}}"
+  with_items:
+    - /opt/organizr
```
Comment (l3uddz) [inline roles/organizr/tasks/main.yml]: missing an empty line below here

File: roles/plex_autoscan/tasks/main.yml [hunk 4]
```diff
@@ -7,7 +7,12 @@
 - name: Stop existing plex_autoscan service
   systemd: state=stopped name=plex_autoscan
   when: plex_autoscan_service.stat.exists
-  
+
+- name: Ensure unionfs_cleaner folders exists
```
Comment (l3uddz) [inline roles/plex_autoscan/tasks/main.yml]: wrong name here

File: roles/plexpy/tasks/main.yml [hunk 5]
```diff
@@ -17,6 +17,10 @@
     path: "/opt/plex/Library/Application Support/Plex Media Server/Logs"
     state: present
 
+- name: Create directories
+  file: "path={{item}} state=directory mode=0775 owner={{user}} group={{user}}"
+  with_items:
+    - /opt/plexpy
```
Comment (l3uddz) [inline roles/plexpy/tasks/main.yml]: missing line below here

File: roles/plex_autoscan/tasks/main.yml (hunk 4 above)
Comment (l3uddz) [reply roles/plex_autoscan/tasks/main.yml]: also the folder has been created yet, but still being created in common.

**Comment by l3uddz** [comment]
Must say, I am a little apprehensive about the static IP removal. It can be very handy in the future that these static IP's exist. As one example, a full installation type has plex and plex_autoscan. plex_autoscan needs to beable to send empty  trash and analyze requests to plex via HTTP. With static ip's this opens the possibility that this can be done locally, e.g. autoscan -> static ip -> plex, whereas without knowing its dynamic ip, you would have todo autoscan -> public ip -> plex, even though they are both local.

**Comment by l3uddz** [comment]
https://github.com/aacebedo/dnsdock

I think this could be a nice solution to the problem above.

**Comment by horjulf** [comment]
@l3uddz if the ports are exposed you can access the services by using the docker gateway IP instead of the container. The ports/services will always be available to all containers through the gateway IP unless explicitly blocked.

**Comment by l3uddz** [comment]
please resolve the conflict and will aim to test+accept this over the weekend.

**Comment by RXWatcher1** [comment]
conflicts fixed

**Comment by l3uddz** [comment]
looks good, will do some testing then accept. thanks!

**Pull request merged**